src/components/CustomDisciplineMenuItem.tsx

```diff
@@ -55,34 +55,42 @@ const CustomDisciplineMenuItem = ({
     return `custom-${id}`
   }
 
-  return loading ? (
-    <Placeholder>
-      <LoadingSpinner>
-        <Loading isLoading={true} />
-      </LoadingSpinner>
-    </Placeholder>
-  ) : data ? (
-    <MenuItem
-      selected={idToSelectedIdString(data.id) === selectedId}
-      onPress={() => {
-        setSelectedId(idToSelectedIdString(data.id))
-        navigation.navigate('DisciplineSelection', {
-          extraParams: { discipline: data }
-        })
-      }}
-      icon={data.icon}
-      title={data.title}>
-      <Description selected={idToSelectedIdString(data.id) === selectedId}>
-        {data.numberOfChildren} {data.numberOfChildren === 1 ? labels.home.unit : labels.home.units}
-      </Description>
-    </MenuItem>
-  ) : (
-    <Placeholder>
-      <ErrorText>
-        {labels.home.errorLoadCustomDiscipline} {apiKey}
-      </ErrorText>
-    </Placeholder>
-  )
+  if (loading) {
+    return (
+      <Placeholder>
+        <LoadingSpinner>
+          <Loading isLoading={true} />
+        </LoadingSpinner>
+      </Placeholder>
+    )
+  } else if (data) {
+    return (
+      <Loading isLoading={loading}>
```

Comment:
I think this is redundant since you are checking for `loading` above already, right? One of both should probably be removed.

Reply:
removed it.